electionleaflets/apps/parties/views.py

```diff
@@ -52,6 +31,10 @@ def get_context_data(self, **kwargs):
             | Q(ynr_party_id=self.kwargs["pk"])
             | Q(ynr_party_id=f"party:{id}")
         )
+        if not qs.exists():
+            raise Http404
```

Comment:
I think you need to raise `Http404()` or `Http404("message")` here. You can't throw a class

Reply:
I added some tests for this, because clearly it needed them. I'd previously manually tested. The answer is: better to raise the instance of the exception, but it works either way!